File: AdopcionesGonzalo/mascota/src/mascota/entities/mascota.entity.ts
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn, OneToMany } from 'typeorm';

@Entity('Mascota')
export class Mascota {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column()
  especie: string;

  @Column()
  raza: string;

  @Column({ type: 'int' })
  edad: number;

  @Column()
  genero: string;

  @Column()
  descripcion: string;

  @Column()
  foto_url: string;

  @Column({ default: true })
  estado_adopcion: boolean;

}

File: mascotasGinger/mascotas/src/main/java/com/mascotas/mascotas/infrastructure/persistence/MascotaEntity.java
package com.mascotas.mascotas.infrastructure.persistence;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "mascota")
public class MascotaEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String especie;
    private String raza;
    private int edad;
    private String sexo;
    private String color;
    @Column(name = "fecha_ingreso")
    private LocalDate fechaIngreso;
} 